lib/scope.js

```diff
@@ -411,6 +432,152 @@ module.exports = {
     return winner;
   },  // Ends scope.tapElement()
 
+  // Handle server reload errors
+  server_statuses: [],
+  server_check_interval_ms: 2 * 1000,
+  stop_tracking_server_status: false,
+  track_server_status: async function( scope ) {
+    /** Updates server_statuses list by checking if server is responding until,
+    *    eventually, it is told to stop. */
+    // This code assumes a reasonable response time would be 2 seconds if the
+    // server is up and that a server reload/restart won't take less than 2 seconds.
+    // Also, it's non-blocking, so it doesn't slow down tests
+    let is_up = await server.is_server_responding({ timeout: scope.server_check_interval_ms });
+    // Add status to the beginning of the list to make truncating easier
+    scope.server_statuses.unshift( is_up );
+
+    // To make it easier to check past server statuses, remove defunct server statuses
+    let max_length = Math.round( scope.timeout/scope.server_check_interval_ms );
+    // Trying to extract more items than the actual length is ok
+    scope.server_statuses = scope.server_statuses.slice( 0, max_length );
+
+    // Unless told to stop, store the next status
+    if ( !scope.stop_tracking_server_status ) {
+      // No await. That's already happened with the server response await. I think.
+      // Will this cause the process to not exit? If so, only for a short time.
+      scope.track_server_status( scope );
```

Comment:
Hmm, I was worried that a while loop would block execution, but it seems to not do that (probably because it's inside an async function), so a while loop it will be!